addons/stock_account/migrations/8.0.1.1/post-migrate.py

```diff
@@ -42,7 +44,65 @@ def create_properties(cr, pool):
                                context=ctx)
 
 
-@openupgrade.migrate()
+def propagate_invoice_state(cr):
+    """ Invoice state is now propagated from sale order to procurement to
+    stock move and picking. We trace it back from the picking and update
+    stock moves and procurements. """
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE stock_move sm
+        SET invoice_state = (
+            SELECT sp.invoice_state FROM stock_picking sp
+            WHERE sm.picking_id = sp.id)
+        WHERE picking_id IN (
+            SELECT id FROM stock_picking sp
+            WHERE sm.picking_id = sp.id
+            AND sm.invoice_state <> sp.invoice_state)
+    """)
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE procurement_order po
+        SET invoice_state = sm.invoice_state
+        FROM stock_move sm
+        WHERE sm.procurement_id = po.id
+    """)
+
+
+def inventory_period_id(cr, pool):
+    period_obj = pool['account.period']
```

Comment:
Document this method putting something like "Fill account period in inventories."